### PR TITLE
Fix: log pipelines: generated operators should have appropriate if condition to avoid spamming collector logs

### PR DESCRIPTION
**Pull request by raj-k-singh**
Fixes https://github.com/SigNoz/signoz/issues/3793

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]
My understanding of parser part is very limited. I mainly looked at the simulator part. Please get reviews from Nitya also.

File: pkg/query-service/app/logparsingpipeline/pipelineBuilder.go [hunk 1]
```diff
@@ -73,7 +76,28 @@ func getOperators(ops []PipelineOperator) []PipelineOperator {
...
+					`%s matches "%s"`,
+					operator.ParseFrom,
+					strings.ReplaceAll(
+						strings.ReplaceAll(operator.Regex, `\`, `\\`),
+						`"`, `\"`,
+					),
+				)
+
+			} else if operator.Type == "json_parser" {
+				operator.If = fmt.Sprintf(`%s matches "\\s*{.*}\\s*$"`, operator.ParseFrom)
```
Comment (nityanandagohain) [inline pkg/query-service/app/logparsingpipeline/pipelineBuilder.go:90]: This is the regex I use for json `body matches "^{.*}$"`

Comment (nityanandagohain) [reply pkg/query-service/app/logparsingpipeline/pipelineBuilder.go]: cool.

**Review by srikanthccv — APPROVED** [review]

**Review by nityanandagohain — APPROVED** [review]
LGTM! I have tested this by creating a nginx parser and application parser, here is the result it generated.

```
    logstransform/pipeline_nginxparser:
        operators:
            - type: router
              id: router_signoz
              routes:
                - output: nginxregexx
                  expr: attributes.source == "nginx"
              default: noop
            - type: regex_parser
              id: nginxregexx
              on_error: send
              if: body != nil && body matches "(?P<remote_addr>((?:[0-9]{1,3}\\.){3}[0-9]{1,3})) (\\S+) (?P<remote_user>\\S+) \\[(?P<time_local>[\\w:\\/]+\\s[+|-]\\d{4})\\] \\\"(?P<request>\\S+)\\s?(?P<path>\\S+)?\\s?(?P<protocol>\\S+)?\\\" (?P<status_code>\\d{3}|-) (?P<body_bytes_sent>\\d+|-)\\s?\\\"?(?P<http_referer>[^\\\"]*)\\\"?\\s\\\"?(?P<http_user_agent>[^\\\"]*)\\\""
              parse_to: attributes
              regex: (?P<remote_addr>((?:[0-9]{1,3}\.){3}[0-9]{1,3})) (\S+) (?P<remote_user>\S+) \[(?P<time_local>[\w:\/]+\s[+|-]\d{4})\] \"(?P<request>\S+)\s?(?P<path>\S+)?\s?(?P<protocol>\S+)?\" (?P<status_code>\d{3}|-) (?P<body_bytes_sent>\d+|-)\s?\"?(?P<http_referer>[^\"]*)\"?\s\"?(?P<http_user_agent>[^\"]*)\"
              parse_from: body
            - type: noop
              id: noop
    logstransform/pipeline_parseapplicationlogs:
        operators:
            - type: router
              id: router_signoz
              routes:
                - output: jsonparser
                  expr: attributes.source == "app"
              default: noop
            - type: json_parser
              id: jsonparser
              output: moveservice
              if: body != nil && body matches "^\\s*{.*}\\s*$"
              parse_to: attributes.temp
              parse_from: body
            - type: move
              id: moveservice
              output: movemethod
              if: attributes?.temp?.service != nil
              from: attributes.temp.service
              to: attributes.service
            - type: move
              id: movemethod
              output: movebody
              if: attributes?.temp?.method != nil
              from: attributes.temp.method
              to: attributes.method
            - type: move
              id: movebody
              output: removetemp
              if: attributes?.temp?.body != nil
              from: attributes.temp.body
              to: body
            - type: remove
              id: removetemp
              if: attributes?.temp != nil
              field: attributes.temp
            - type: noop
              id: noop
 ````
 
 Which is a working state

**Pull request merged**